### ISSUE TITLE
How can I get issue type error/warning?

### ISSUE THREAD

**Issue by xpol** [mpeterv/luacheck#31]
Currently linter-luacheck use the last line to determine the issue type:

If i got:

    Checking source.lua ....
    
        source.lua:136:10: ....
        source.lua:137:16: ....
        source.lua:156:20: ....
    
    Total: x warnings / y errors in 1 file

The errors number `y` is `0`, I think all issue reported is warnings.
Else if `y` is not `0` i think all issues are errors.

I think this way is not the best solution.

Can we have each report line like:

        source.lua:136:10: error: ....
        source.lua:136:10: warning: ....

So that I can parse the issue type (error or warning) from each line itself.

I see there is `format_warning` and `format_error_msg` we can add `warning:` and  `error:` in the 2 function.

**Comment by xpol** [comment]
I mean does `luacheck` always have one of warning count and error count to be zero?

**Comment by mpeterv** [comment]
> I mean does luacheck always have one of warning count and error count to be zero?

When checking only one file, yes. I agree that it's better to have some indication on each line. When using --codes option most warnings have a label with warning code near them starting with `W`, e,g, `source.lua:136:10: (W212) unused argument baz`, you could check for that.

Perhaps it'll make even more sense to give codes to errors as well, so that a syntax error is reported like this: `source.lua:2:1: (E011) expected '=' near <eof>`. What do you think?

**Comment by xpol** [comment]
:+1:  It would be nice that every warnings and errors has a codes.

And then `format_warning` and `format_error_msg` should have same logic and merged as one function.

**Comment by mpeterv** [comment]
On master errors now also have codes. Just run luacheck with `--codes` option and then for each issue look for code inside parens, it will start with `E` for errors and with `W` for warnings, e.g.

```
Checking spec/samples/python_code.lua             1 error

    spec/samples/python_code.lua:1:6: (E011) expected '=' near '__future__'

Checking spec/samples/inline_options.lua          8 warnings / 2 errors

    spec/samples/inline_options.lua:6:16: (W211) unused function 'f'
    spec/samples/inline_options.lua:12:4: (W113) accessing undefined variable 'qu'
    spec/samples/inline_options.lua:15:1: (W113) accessing undefined variable 'baz'
    spec/samples/inline_options.lua:24:10: (W211) unused variable 'g'
    spec/samples/inline_options.lua:26:7: (W211) unused variable 'f'
    spec/samples/inline_options.lua:26:10: (W211) unused variable 'g'
    spec/samples/inline_options.lua:28:1: (E022) unpaired push directive
    spec/samples/inline_options.lua:30:4: (E023) unpaired pop directive
    spec/samples/inline_options.lua:36:1: (W541) empty do..end block
    spec/samples/inline_options.lua:37:10: (W542) empty if branch

Total: 8 warnings / 3 errors in 2 files
```

**Comment by xpol** [comment]
@mpeterv :+1: thank you!
